practices/managing-node-projects.md

```diff
@@ -18,54 +24,80 @@
+### `npm install`
 
-### Run `nvm use` before `npm install`
+`npm install` is the most common method, but it doesn't guarantee reproducible/consistent builds. npm can install varying versions of dependencies, according to [semantic versioning](https://docs.npmjs.com/files/package.json#dependencies) and the version of node used when installing. `npm install` will create a `package-lock.json` file describing the complete dependency graph and specifying exact versions of all the dependencies installed.
 
-You should always ensure you're using the correct version of node (and implicitly `npm`) before doing an `npm install`.
+Note that if a `package-lock.json` is present **`npm install` will not use it** as the source of authority for what to install.
 
-Running `nvm use` before `npm install` is good because:
-1. It make installs _more_ predictable, which minimises "but it works on my machine" issues.
-1. It minimises changes to the `package-lock.json` (if your project is comitting that file).
+This is the method we recommend for installing dependencies when authoring node _libraries_.
 
-A problem with using `npm install` is that it doesn't guarantee reproducible builds, as neither the `package.json` nor `package-lock.json` is a source of authority for what's installed.
+### `npm ci`
 
-Better is to use a version of node that ships with `npm` version 5.7.0 or higher ([node v10.3+ or 8.12+](https://nodejs.org/en/download/releases/)). This is because `npm` 5.7.0+ supports the `ci` argument, which is good because **`npm ci` will always install predictably**, using the `package-lock.json` as a source of authority.
+`npm ci` installs the complete dependency graph exactly as specified in the `package-lock.json` file, and so guarantees reproducible/consistent builds. This helps reduce "but it works on my machine" issues and is suitable for CI/CD environments (as the name implies).
 
 [`npm ci` is also much quicker than `npm install`](https://docs.npmjs.com/cli/ci.html#description) if the `./node_modules` directory is not present (such as in a CI environment).
 
-(You could use `npm ci` by specifying a newer version of `npm` than is recommended for your [particular version of node](https://nodejs.org/en/download/releases/), but `npm` is itself written in node and [only supports certain node versions](https://github.com/npm/cli/blob/latest/lib/utils/unsupported.js).)
+This is the method we recommend for installing dependencies when authoring node _applications_.
+
+### Using the right version of node
+
+When authoring libraries you're probably supporting multiple versions of node, so you're not required to use a specific version when installing (as long as the version is supported by your library).
+
+When authoring node applications, and therefore using `npm ci`, it is important to specify a version of node to use. This is because the `package-lock.json` file will reflect the version of node used when the `package-lock.json` file was created.
```

Comment:
I think it is true but am having a hard time finding anything beyond anecdata and random SO posts, so it may well be a superstition :)

I'm sure I remember problems with node-sass in this area, and fsevents, but I think these are examples where it _will or wont install_ according to _platform_. It's also true of course that packages may or may not install due to the user not using a supported version of node. But these are examples of a package either installing or not, and not examples of a successful install but of different _versions_ of the package.

Another thing to support that theory is that I cannot see a way to target different versions of packages at different versions of node.

So I think I should amend this line, and the following line, thanks!

Reply:
Have removed this section as I think its pointless.